### PR TITLE
Add hist_bucket_to_string method

### PR DESCRIPTION
**Pull request by HeinrichHartmann**
Hit some problems with NaN handing on the way, and cleaned that up.

### CODE REVIEW

File: src/circllhist.h [hunk 1]
```diff
@@ -64,6 +64,23 @@ API_EXPORT(double) hist_bucket_to_double_bin_width(hist_bucket_t hb);
...
+//! - t = '+' or '-' exponent sign
+//! - yyy -- three digits representing exp as decimal integer with leading 0s
+//!
+//! Exception: The zero bucket is represented  as "0"
+//! Exception: Invalid buckets are represented as "NAN"
+//!
+//! Examples:
+//!      1  => '+10e-001'; 12    => '+12e+000';
+//!  -0.23  => '-23e-003'; 23000 => '+23e+003';
+API_EXPORT(void) hist_bucket_to_string(hist_bucket_t hb, char *buf);
```
Comment (aidancully) [inline src/circllhist.h]: please add something like
```c
//! Size of standard-format string bucket size.
#define CIRCLL_BUCKET_STR_SIZE 8
```

Comment (aidancully) [inline src/circllhist.h]: (fwiw, I'm not sure if that's the best name... maybe `HIST_BUCKET_STR_LEN` instead?)

Comment (HeinrichHartmann) [inline src/circllhist.h]: I can do that, but note that this constant appears no-where inside the code.
The buffer is filed by explicitly referencing bytes. No iteration occurs.

File: src/circllhist.c [hunk 2]
```diff
@@ -369,8 +374,8 @@ int hist_bucket_cmp(hist_bucket_t h1, hist_bucket_t h2) {
   // checks if h1 < h2 on the real axis.
   if(*(uint16_t *)&h1 == *(uint16_t *)&h2) return 0;
   /* place NaNs at the beginning always */
-  if(h1.val == (int8_t)0xff) return 1;
-  if(h2.val == (int8_t)0xff) return -1;
+  if(h1.val == -1) return 1;
+  if(h2.val == -1) return -1;
```
Comment (postwait) [inline src/circllhist.c]: Why this change?  The val with all it's bits set is NaN. -1 is a tad less clear.

File: src/circllhist.c [hunk 3]
```diff
@@ -928,3 +931,24 @@ hist_compress_mbe(histogram_t *hist, int8_t mbe) {
   }
   return hist_compressed;
 }
+
+extern void
+hist_bucket_to_string(hist_bucket_t hb, char *buf) {
+  // hb_nan would land in 0-bucket if we not catch it fist
+  if(hb.val == hb_nan.val && hb.exp == hb_nan.exp) strcpy(buf, "NAN");
+  else if(hb.val < -99 || 99 < hb.val) strcpy(buf, "NAN");
```
Comment (postwait) [inline src/circllhist.c]: Can we use NaN? instead of NAN... in other language specs this is more common.

**Comment by HeinrichHartmann** [comment]
squashed updates.
